internal/kafka/internal/handlers/validation.go

```diff
@@ -37,9 +38,16 @@ const minimunNumberOfNodesForTheKafkaMachinePool = 3
 
 func validateKafkaBillingModel(ctx context.Context, kafkaService services.KafkaService, kafkaConfig *config.KafkaConfig, kafkaRequestPayload *public.KafkaRequestPayload) handlers.Validate {
 	return func() *errors.ServiceError {
+		billingModel := shared.SafeString(kafkaRequestPayload.BillingModel)
+		// enterprise kafkas billing model validation
+
+		if !shared.StringEmpty(kafkaRequestPayload.ClusterId) && !shared.StringEqualsIgnoreCase(billingModel, string(v1.BillingModelStandard)) {
+			return errors.InvalidBillingAccount("invalid billing model: %s, only %v is allowed", billingModel,
+				v1.BillingModelStandard)
```

Comment:
the billing mode has to be standard https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/ba3f5f14f64443a12a20a2df6d38e045b3637054/config/kafka-instance-types-configuration.yaml#L82

Reply:
I also think we should validate that, once the billing model is `enterprise` - then a cluster_id has to be given.

Reply:
`the billing mode has to be standard` - you mean enterprise? 

`I also think we should validate that, once the billing model is enterprise - then a cluster_id has to be given.` - isn't it already done here `if !shared.StringEmpty(kafkaRequestPayload.ClusterId)`?

Reply:
> `the billing mode has to be standard` - you mean enterprise?

Sorry, yes. It has to be enterprise.
 
> `I also think we should validate that, once the billing model is enterprise - then a cluster_id has to be given.` - isn't it already done here `if !shared.StringEmpty(kafkaRequestPayload.ClusterId)`?

I can see the condition but I do not think if it covers the case where: someone attempts to create a Kafka with the enterprise billing but they've not supplied a clusterID.
 
What I am proposing is to add a check if `billingModel == "enterprise" && clusterID == "" then error("You must supply a cluster id")`

Reply:
changed

Reply:
Thanks